openedx/core/djangoapps/user_api/helpers.py

```diff
@@ -255,7 +255,12 @@ def add_field(
         # Any field property can be overwritten (for example, the default value or placeholder)
         field_dict.update(field_override)
 
-        self.fields.append(field_dict)
+        if field_dict['name'] == 'confirm_email':
```

Comment:
What is the purpose of this check?

Reply:
> What is the purpose of this check?

This makes sure that confirm-email is next to email field on the registration form otherwise it will be rendered at different position.